instat/dlgRugPlot.vb

```diff
@@ -125,7 +102,28 @@ Public Class dlgRugPlot
         clsBaseOperator.SetAssignTo("last_graph", strTempDataframe:=ucrRugPlotSelector.ucrAvailableDataFrames.cboAvailableDataFrames.Text, strTempGraph:="last_graph")
         ucrBase.clsRsyntax.SetBaseROperator(clsBaseOperator)
         bResetSubdialog = True
+    End Sub
 
+    Public Sub SetRCodeForControls(bReset As Boolean)
+        ucrSaveGraph.SetRCode(clsBaseOperator, bReset)
+        ucrRugPlotSelector.SetRCode(clsDefaultFunction, bReset)
+
+        ucrReceiverX.SetRCode(clsRaesFunction, bReset)
+        ucrFactorOptionalReceiver.SetRCode(clsRaesFunction, bReset)
+        ucrVariablesAsFactorForRugPlot.SetRCode(clsRaesFunction, bReset)
+    End Sub
+
+    Private Sub TestOkEnabled()
+        ' Tests when OK is enabled
+        If (Not ucrSaveGraph.IsComplete) OrElse (ucrVariablesAsFactorForRugPlot.IsEmpty AndAlso ucrReceiverX.IsEmpty()) Then

```

Comment:
I noticed on previous versions OK was always enabled because it was commented out and you made it back into the code. However, the R code does run (as it did with dlgScatterplot before) to have either ucrVariablesAsFactor or ucrReceiverX (or both) full. So, i changed it slightly.
Perhaps look out for this on future graphics dialogs?